satellite/satellitedb/nodeselection.go

```diff
@@ -90,12 +91,20 @@ func (cache *overlaycache) SelectStorageNodes(ctx context.Context, totalNeededNo
 	return nodes, nil
 }
 
-func (cache *overlaycache) selectStorageNodesOnce(ctx context.Context, reputableNodeCount, newNodeCount int,
-	criteria *overlay.NodeCriteria, excludedIDs []storj.NodeID,
-	excludedNetworks []string) (reputableNodes, newNodes []*overlay.SelectedNode, err error) {
-
+func (cache *overlaycache) selectStorageNodesOnce(ctx context.Context, reputableNodeCount, newNodeCount int, criteria *overlay.NodeCriteria, excludedIDs []storj.NodeID, excludedNetworks []string) (reputableNodes, newNodes []*overlay.SelectedNode, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	var asOf string
+	var asOfSystemTimeInterval time.Duration
+
+	asOfSystemTimeEnabled := cache.db.implementation == dbutil.Cockroach && criteria.AsOfSystemTimeInterval != 0
+
+	if asOfSystemTimeEnabled {
+		asOf = fmt.Sprintf(" AS OF SYSTEM TIME '%s'", criteria.AsOfSystemTimeInterval.String())
```

Comment:
This string is created in multiple places, have a single place that defines proper handling, so there's less duplication.